plugin/advisor/mysql/mysql_column_requirement_advisor.go

```diff
@@ -0,0 +1,196 @@
+	advisor.Register(db.TiDB, advisor.MySQLColumnRequirement, &ColumnRequirementAdvisor{})
+}
+
+// ColumnRequirementAdvisor is the advisor checking for column requirement.
+type ColumnRequirementAdvisor struct {
+}
+
+// Check checks for the column requirement.
+func (adv *ColumnRequirementAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+	payload, err := api.UnmarshalRequiredColumnRulePayload(ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	requiredColumns := make(columnSet)
+	for _, column := range payload.ColumnList {
+		requiredColumns[column] = true
+	}
+	checker := &columnRequirementChecker{
+		level:              level,
+		requiredColumnList: strings.Join(payload.ColumnList, ", "),
+		requiredColumns:    requiredColumns,
+		tables:             make(tableState),
+	}
+
+	for _, stmtNode := range root {
+		(stmtNode).Accept(checker)
+	}
+
+	return checker.generateAdvisorList(), nil
+}
+
+type columnSet map[string]bool
+type tableState map[string]columnSet
+
+type columnRequirementChecker struct {
+	advisorList        []advisor.Advice
+	level              advisor.Status
+	requiredColumns    columnSet
+	requiredColumnList string
+	tables             tableState
+}
+
+func (v *columnRequirementChecker) Enter(in ast.Node) (ast.Node, bool) {
+	switch node := in.(type) {
+	// CREATE TABLE
+	case *ast.CreateTableStmt:
+		v.createTable(node)
+	// ALTER TABLE
+	case *ast.AlterTableStmt:
+		for _, spec := range node.Specs {
+			table := node.Table.Name.O
+			switch spec.Tp {
+			// RENAME COLUMN
+			case ast.AlterTableRenameColumn:
+				v.renameColumn(table, spec.OldColumnName.Name.O, spec.NewColumnName.Name.O)
+			// ADD COLUMNS
+			case ast.AlterTableAddColumns:
+				for _, column := range spec.NewColumns {
+					v.addColumn(table, column.Name.Name.O)
+				}
+			// DROP COLUMN
+			case ast.AlterTableDropColumn:
+				v.dropColumn(table, spec.OldColumnName.Name.O)
+			// CHANGE COLUMN
+			case ast.AlterTableChangeColumn:
+				v.renameColumn(table, spec.OldColumnName.Name.O, spec.NewColumns[0].Name.Name.O)
+			}
+		}
+	}
+	return in, false
+}
+
+func (v *columnRequirementChecker) Leave(in ast.Node) (ast.Node, bool) {
+	return in, true
+}
+
+func (v *columnRequirementChecker) generateAdvisorList() []advisor.Advice {
+	for tableName, table := range v.tables {
+		for column := range v.requiredColumns {
+			if exist, ok := table[column]; !ok || !exist {
+				v.advisorList = append(v.advisorList, advisor.Advice{
+					Status:  v.level,
+					Code:    common.NoRequiredColumn,
+					Title:   "Require columns",
+					Content: fmt.Sprintf("%q requires columns: %s", tableName, v.requiredColumnList),
```

Comment:
Can we just use the `requiredColumns` set?

Reply:
I implement the function `columnSet.String()` to print column list.
Because in Golang, traversing the `map` is in random order. Details in [Go maps in action
](https://go.dev/blog/maps)